src/components/AlertRuleForm.tsx

```diff
@@ -139,9 +140,10 @@ const getStyles = (theme: GrafanaTheme) => ({
 type Props = {
   rule: AlertRule;
   onSubmit: (alertValues: AlertFormValues) => Promise<FetchResponse<unknown> | undefined>;
+  children: React.ReactNode;
 };
 
-export const AlertRuleForm = ({ rule, onSubmit }: Props) => {
+export const CommonAlertForm = ({ rule, onSubmit, children }: Props) => {
```

Comment:
You might bump into some weird behavior here. You are creating a form and all it's state management and context and whatnot here in CommonAlertForm on line 151, but doing it again inside the child component (like on line 287). I think one or the other is fine, but doing both might result in something unexpected happening

Reply:
I think leaving it up to the children is probably best?